plugins/mirror/bazel.go

```diff
@@ -24,6 +25,9 @@ type Artifact struct {
 }
 
 func (a *Artifact) URLs() []string {
+	if a.rule.AttrString("url") != "" {
```

Comment:
I think you have a typo here and on a few other places. We are always looking for `urls` (plural).

Reply:
The failure occured because of an element that had no `urls` attribute, but where a `url` element as present. I added the handling for that case.